pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -77,139 +73,161 @@ func New(args runtime.Object, handle frameworktypes.Handle) (frameworktypes.Plug
-	if err != nil {
-		return nil, fmt.Errorf("could not get selector from label selector")
+	return nil
+}
+
+var AllPolicies = []PodProtectionPolicy{
+	PodsWithLocalStorage,
+	DaemonSetPods,
+	SystemCriticalPods,
+	FailedBarePods,
+	PodsWithPVC,
+	PodsWithoutPDB,
+}
+
+// applyEffectiveProtectedPolicies applies all relevant protection policies to the plugin.
+// For each policy:
+// - If it's in the list, we disallow eviction (shouldEvict = false).
+// - If it's NOT in the list, we allow eviction (shouldEvict = true).
+func applyEffectiveProtectedPolicies(ev *DefaultEvictor, policies []PodProtectionPolicy, handle frameworktypes.Handle) error {
+	policyMap := make(map[PodProtectionPolicy]bool)
+	for _, policy := range policies {
+		policyMap[policy] = true
 	}
-	if defaultEvictorArgs.LabelSelector != nil && !selector.Empty() {
-		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if !selector.Matches(labels.Set(pod.Labels)) {
-				return fmt.Errorf("pod labels do not match the labelSelector filter in the policy parameter")
-			}
-			return nil
-		})
+
+	for _, policy := range AllPolicies {
+		shouldEvict := !policyMap[policy]
+
+		switch policy {
+		case PodsWithLocalStorage:
+			addEvictionConstraintsForLocalStoragePods(ev, shouldEvict)
+		case DaemonSetPods:
+			addEvictionConstraintsForDaemonSetPods(ev, shouldEvict)
+		case SystemCriticalPods:
+			addEvictionConstraintsForSystemCriticalPods(ev, shouldEvict, handle, ev.args.PriorityThreshold)
+		case FailedBarePods:
+			addEvictionConstraintsForFailedBarePods(ev, shouldEvict)
+		case PodsWithPVC:
+			addEvictionConstraintsForPvcPods(ev, shouldEvict)
+		case PodsWithoutPDB:
+			addEvictionConstraintsForPodsWithoutPDB(ev, shouldEvict, handle)
+		default:
+			return fmt.Errorf("unknown protection policy: %s", string(policy))
+		}
 	}
 
-	if defaultEvictorArgs.MinReplicas > 1 {
-		indexName := "metadata.ownerReferences"
-		indexer, err := getPodIndexerByOwnerRefs(indexName, handle)
-		if err != nil {
-			return nil, err
+	return nil
+}
+
+// contains checks if a slice of PodProtectionPolicy contains a given policy.
+func contains(list []PodProtectionPolicy, item PodProtectionPolicy) bool {
+	for _, v := range list {
+		if v == item {
+			return true
 		}
-		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
-			if len(pod.OwnerReferences) == 0 {
-				return nil
-			}
+	}
+	return false
+}
 
-			if len(pod.OwnerReferences) > 1 {
-				klog.V(5).InfoS("pod has multiple owner references which is not supported for minReplicas check", "size", len(pod.OwnerReferences), "pod", klog.KObj(pod))
-				return nil
-			}
+// getEffectiveProtectedPolicies determines which policies are currently active.
+// It supports both new-style (PodProtectionPolicies) and legacy-style flags.
+func getEffectiveProtectedPolicies(args *DefaultEvictorArgs) []PodProtectionPolicy {
+	if args.defaultPodProtectionPolicies == nil {
+		args.defaultPodProtectionPolicies = BuiltInPodProtectionPolicies
```

Comment:
`defaultPodProtectionPolicies` is used only for internal purposes so the same list of default policies does not have to be duplicated. If you plan to define `BuiltInPodProtectionPolicies` as well the new variable can be used instead of the internal `defaultPodProtectionPolicies` field.